kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/UserTaskEscalationIntegrationTest.java

```diff
@@ -140,6 +140,39 @@ public void testEscalation() throws InterruptedException, MessagingException, Ex
         }
     }
 
+    @Test
+    public void testCompleteTaskBeforeEscalation() throws InterruptedException {
+        assertSuccess(client.createContainer(CONTAINER_ID, new KieContainerResource(CONTAINER_ID, releaseId)));
+
+        Long processInstanceId = processClient.startProcess(CONTAINER_ID, PROCESS_ID_USERTASK_ESCALATION, params);
+        assertNotNull(processInstanceId);
+        assertTrue(processInstanceId > 0);
+
+        List<TaskSummary> taskList = taskClient.findTasksAssignedAsPotentialOwner(USER_YODA, 0, 10);
+        assertNotNull(taskList);
+        assertEquals(1, taskList.size());
+        TaskSummary taskSummary = taskList.get(0);
+        assertEquals("User Task", taskSummary.getName());
+        Long taskId = taskSummary.getId();
+
+        TaskInstance taskInstance = taskClient.findTaskById(taskId);
+        assertNotNull(taskInstance);
+        assertEquals(USER_YODA, taskInstance.getActualOwner());
+
+        taskClient.startTask(CONTAINER_ID, taskId, USER_YODA);
+        taskClient.completeTask(CONTAINER_ID, taskId, USER_YODA, new HashMap<String, Object>());
+
+        ProcessInstance processInstance = processClient.getProcessInstance(CONTAINER_ID, processInstanceId);
+        assertNotNull(processInstance);
+        assertEquals(org.kie.api.runtime.process.ProcessInstance.STATE_COMPLETED, processInstance.getState().intValue());
+
+        assertNullOrEmpty("Email recieved!", wiser.getMessages());
+
+        //wait while, cause email is sended 2s after task start
+        Thread.sleep(3000l);
```

Comment:
Well, in this case we are checking that something didn't happen. There could be made a loop checking mails periodically, to fail faster in case of error.